float32[] x_traj
float32[] y_traj
float32[] theta_traj